client/client.go

```diff
@@ -1719,3 +1734,70 @@ func trimHTTPPrefix(str string) string {
 	str = strings.TrimPrefix(str, "https://")
 	return str
 }
+
+func (c *client) LoadGlobalConfig(ctx context.Context, names []string) ([]GlobalConfigItem, error) {
+	resp, err := c.getClient().LoadGlobalConfig(ctx, &pdpb.LoadGlobalConfigRequest{Names: names})
+	if err != nil {
+		return nil, err
+	}
+	res := make([]GlobalConfigItem, len(resp.Items))
+	for i, item := range resp.Items {
+		cfg := GlobalConfigItem{Name: item.Name}
+		if item.Error != nil {
+			if item.Error.Type == pdpb.ErrorType_GLOBAL_CONFIG_NOT_FOUND {
+				cfg.Error = errGlobalConfigNotFound
+			} else {
+				cfg.Error = errors.New(item.Error.Message)
+			}
+		} else {
+			cfg.Value = item.Value
+		}
+		res[i] = cfg
+	}
+	return res, nil
+}
+
+func (c *client) StoreGlobalConfig(ctx context.Context, items []GlobalConfigItem) error {
+	res_arr := make([]*pdpb.GlobalConfigItem, len(items))
+	for i, it := range items {
+		res_arr[i] = &pdpb.GlobalConfigItem{Name: it.Name, Value: it.Value}
+	}
+	res, err := c.getClient().StoreGlobalConfig(ctx, &pdpb.StoreGlobalConfigRequest{Changes: res_arr})
+	if err != nil {
+		return err
+	}
+	res_err := res.GetError()
+	if res_err != nil {
+		return errors.Errorf(res_err.Message)
+	}
+	return err
+}
+
+func (c *client) WatchGlobalConfig(ctx context.Context) (chan []GlobalConfigItem, error) {
+	receiver := make(chan []GlobalConfigItem, 16)
+	res, err := c.getClient().WatchGlobalConfig(ctx, &pdpb.WatchGlobalConfigRequest{})
+	if err != nil {
+		close(receiver)
+		return nil, err
+	}
+	go func() {
+		defer close(receiver)
+		for {
+			m, err := res.Recv()
+			if err != nil {
+				return
+			}
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				arr := make([]GlobalConfigItem, len(m.Changes))
+				for j, i := range m.Changes {
+					arr[j] = GlobalConfigItem{i.Name, i.Value, nil}
+				}
+				receiver <- arr
```

Comment:
Is there a possibility that the caller of `(*client).WatchGlobalConfig` close the channel from the outside?

Reply:
yes, so i will add an error handling method in this function